src/inference/emcee.jl

```diff
@@ -0,0 +1,243 @@
+        1:spl.alg.ensemble.n_walkers
+    )
+
+    # Convert transitions to array format.
+    # Also retrieve the variable names.
+    params_vec = map(_params_to_array, ts_transform)
+
+    # Extract names and values separately.
+    nms = params_vec[1][1]
+    vals_vec = [p[2] for p in params_vec]
+
+    # Get the values of the extra parameters in each transition.
+    extra_vec = map(get_transition_extras, ts_transform)
+
+    # Get the extra parameter names & values.
+    extra_params = extra_vec[1][1]
+    extra_values_vec = [e[2] for e in extra_vec]
+
+    # Extract names & construct param array.
+    nms = [nms; extra_params]
+    parray = map(x -> hcat(x[1], x[2]), zip(vals_vec, extra_values_vec))
+    parray = cat(parray..., dims=3)
+
+    # Get the average or final log evidence, if it exists.
+    le = getlogevidence(spl)
+
+    # Set up the info tuple.
+    if save_state
+        info = (range = rng, model = model, spl = spl)
+    else
+        info = NamedTuple()
+    end
+
+    # Concretize the array before giving it to MCMCChains.
+    parray = MCMCChains.concretize(parray)
+
+    # Chain construction.
+    return MCMCChains.Chains(
+        parray,
+        string.(nms),
+        deepcopy(TURING_INTERNAL_VARS);
+        evidence=le,
+        info=info,
+        sorted=true
+    )
+end
+
+
+####
+#### Compiler interface, i.e. tilde operators.
+####
+function DynamicPPL.assume(
+    rng,
+    spl::Sampler{<:Emcee},
+    dist::Distribution,
+    vn::VarName,
+    vi,
+)
+    updategid!(vi, vn, spl)
+    r = vi[vn]
+    return r, logpdf_with_trans(dist, r, istrans(vi, vn))
+end
+
+function DynamicPPL.dot_assume(
+    rng,
+    spl::Sampler{<:Emcee},
+    dist::MultivariateDistribution,
+    vn::VarName,
+    var::AbstractMatrix,
+    vi,
+)
+    @assert dim(dist) == size(var, 1)
+    getvn = i -> VarName(vn, vn.indexing * "[:,$i]")
+    vns = getvn.(1:size(var, 2))
+    updategid!.(Ref(vi), vns, Ref(spl))
+    r = vi[vns]
+    var .= r
+    return var, sum(logpdf_with_trans(dist, r, istrans(vi, vns[1])))
+end
+function DynamicPPL.dot_assume(
+    rng,
+    spl::Sampler{<:Emcee},
+    dists::Union{Distribution, AbstractArray{<:Distribution}},
+    vn::VarName,
+    var::AbstractArray,
+    vi,
+)
+    getvn = ind -> VarName(vn, vn.indexing * "[" * join(Tuple(ind), ",") * "]")
+    vns = getvn.(CartesianIndices(var))
+    updategid!.(Ref(vi), vns, Ref(spl))
+    r = reshape(vi[vec(vns)], size(var))
+    var .= r
+    return var, sum(logpdf_with_trans.(dists, r, istrans(vi, vns[1])))
+end
+
+function DynamicPPL.observe(
+    spl::Sampler{<:Emcee},
+    d::Distribution,
+    value,
+    vi,
+)
+    return DynamicPPL.observe(SampleFromPrior(), d, value, vi)
+end
+
+function DynamicPPL.dot_observe(
+    spl::Sampler{<:Emcee},
+    ds::Union{Distribution, AbstractArray{<:Distribution}},
+    value::AbstractArray,
+    vi,
+)
+    return DynamicPPL.dot_observe(SampleFromPrior(), ds, value, vi)
+end
```

Comment:
I guess this might not be needed at all if we never run the `model` with `Sampler{<:EMCEE}` - I thought it should be sufficient to just always use `SampleFromPrior`? I guess one mainly has to change the construction of the `AMH.DensityModel` and pass `SampleFromPrior` instead of `spl` there.